lib/feature_management.rb

```diff
@@ -116,4 +122,56 @@ def self.voip_allowed_phones
       allowed_phones.map { |p| Phonelib.parse(p).e164 }.to_set
     end
   end
+
+  # Whether we collect device profiling information as part of the proofing process.
+  def self.proofing_device_profiling_collecting_enabled?
+    case IdentityConfig.store.proofing_device_profiling
+      when :enabled, :collect_only then true
+      when :disabled then false
+      # BEGIN temporary transitional fallback
+      when nil
+        if IdentityConfig.store.proofing_device_profiling_collecting_enabled.nil?
+          false
+        else
+          IdentityConfig.store.proofing_device_profiling_collecting_enabled
+        end
+      # END temporary transitional fallback
+      else
+        raise 'Invalid value for proofing_device_profiling'
+    end
+  end
+
+  # Whether we prevent users from proceeding with identity verification based on the outcomes of
+  # device profiling.
+  def self.proofing_device_profiling_decisioning_enabled?
+    case IdentityConfig.store.proofing_device_profiling
+    when :enabled then true
+    when :collect_only, :disabled then false
+    # BEGIN temporary transitional fallback
+    when nil
+      if IdentityConfig.store.lexisnexis_threatmetrix_required_to_verify.nil?
+        false
+      else
+        IdentityConfig.store.lexisnexis_threatmetrix_required_to_verify
+      end
+    # END temporary transitional fallback
+    else
+      raise 'Invalid value for proofing_device_profiling'
+    end
+  end
+
+  def self.threatmetrix_mock_enabled?
+    domain_name = IdentityConfig.store.domain_name
+    is_not_allowed = ENVS_WHERE_THREATMETRIX_MOCK_NOT_ALLOWED.include? domain_name
+    return false if is_not_allowed
+
+    tm_config_present = [
+      :lexisnexis_threatmetrix_api_key,
+      :lexisnexis_threatmetrix_base_url,
+    ].any? { |key| IdentityConfig.store.send(key).present? }
```

Comment:
1. do we want `.any?` or all?
2. Since it's a struct we can use `[]` access instead of having to send:

```suggestion
    tm_config_present = [
      :lexisnexis_threatmetrix_api_key,
      :lexisnexis_threatmetrix_base_url,
    ].any? { |key| IdentityConfig.store[key].present? }
```

Reply:
1. I initially had this as `all` but switched to `any` because I'm mostly looking for a signal that you're _trying_ to configure this locally--in that case, your TM API calls would fail and I think that'd be a "better" error than still seeing the mock (that is, it's an error that's closer to the thing that's actually wrong). But I do not have super strong feelings about this.
2. Agreed, makes sense to me

Reply:
1: seems reasonable to keep `.any?` 👍